src/ophys_etl/modules/video/utils.py

```diff
@@ -30,13 +28,14 @@ def create_downsampled_video(
         input_hz: float,
         video_path: pathlib.Path,
         output_hz: float,
-        kernel_size: Optional[int],
+        spatial_filter: Optional[Callable[[np.ndarray], np.ndarray]],
```

Comment:
I actually like that these methods accept callables. It means that, outside of the CLI, we can quickly experiment with new spatial filters by just passing in a function without having to implement the `if/else` block necessary to parse a string into a function.

Again: open to discussion, but I do see this as a feature, rather than a bug.

Reply:
Sure, then you allow this function to accept string or callable. If string, it converts it into a callable.
This would not force the user to go through the CLI to be able to properly pull the correct function for "mean" or "median" filters and would allow your use case of passing a custom callable for experimenting.

Reply:
I'm going to leave this. Your proposal would require adding a `kernel_size` arg to `create_downsampled_video_h5`. I'm not a huge fan of increasing the number of arguments more than we have to. When/if we move to class-based video manipulation, I suspect the `apply_spatial_filter()` method will be general enough to accept an object that is a spatial filter (I know I'm talking like someone who lives in `C++`-ville; in python, it would just be a callable). I still like the generalizability of that. We can revisit this conversation when we move to class-based manipulation of videos.